src/utils/ServerTime.js

```diff
@@ -0,0 +1,60 @@
+import { getUTCEpoch, getLocalDate, getUTCDate } from './index';
+
+class ServerTime {
+    clockStarted = false;
+
+    constructor(api) {
+        this._api = api;
+        this.init();
+    }
+
+    init() {
```

Comment:
we can make `ServerTime` a singleton by passing `api` into `init` instead. This way we don't have to pass a `ServerTime` instance around. It's interesting to note that in their original implementation, binary-static returns undefined if the server time has not been retrieved.

I think we can get around this by making `init` async, and throwing an Error if we try to retrieve the time before the server has responded.